src/Integration.UnitTests/LocalServices/ProjectSystemHelperTests.cs

```diff
@@ -145,6 +184,43 @@ public void ProjectSystemHelper_GetFilteredSolutionProjects()
             CollectionAssert.AreEqual(new[] { vbProject }, actual,
                 "Unexpected projects: {0}", string.Join(", ", actual.Select(p => p.Name)));
         }
+
+        [TestMethod]
+        public void ProjectSystemHelper_AddFileToProject_ArgChecks()
+        {
+            Exceptions.Expect<ArgumentNullException>(() => this.testSubject.AddFileToProject(null, "file"));
+            Exceptions.Expect<ArgumentNullException>(() => this.testSubject.AddFileToProject(new ProjectMock("project"), null));
+            Exceptions.Expect<ArgumentNullException>(() => this.testSubject.AddFileToProject(new ProjectMock("project"), ""));
+            Exceptions.Expect<ArgumentNullException>(() => this.testSubject.AddFileToProject(new ProjectMock("project"), "\t\n"));
+        }
+
+        [TestMethod]
+        public void ProjectSystemHelper_AddFileToProject()
+        {
+            // Setup
+            ProjectMock project = this.solutionMock.AddOrGetProject("project1");
+            string fileToAdd = @"x:\myFile.txt";
+            
+            // Case 1: file not in project
+            // Act
+            this.testSubject.AddFileToProject(project, fileToAdd);
+
+            // Verify
+            Assert.IsTrue(project.Files.ContainsKey(fileToAdd));
+
+            // Case 2: file already in project
+            // Act
+            this.testSubject.AddFileToProject(project, fileToAdd);
+
+            // Verify
+            Assert.IsTrue(project.Files.ContainsKey(fileToAdd));
```

Comment:
Yes, it's more about make sure we don't crash. There are separate unit tests for IsFileInProject and the underlying collection is mocked, so we're quite limited in terms of testing real product and not the mocked classes.